app/models/oral_history_content.rb

```diff
@@ -64,6 +66,12 @@ def has_ohms_index?
     @has_ohms_index ||= ohms_xml&.index_points&.present?
   end
 
+  def set_combined_audio_derivatives_creation_status(status_string)
+    self.combined_audio_derivatives_creation_status = status_string
+    self.combined_audio_derivatives_creation_status_changed_at = DateTime.now
+    self.save!
+  end
```

Comment:
Let's not use any arbitrary string for this, including things like `"Error: #{e}"`. 

Let's use a controlled vocabulary that can only be a limited set of things. I would imagine `queued`, `in_progress`, `success`, and `failure`.  (**Not sure if you are currently recording/tracking "queued", but we should** Should be able to override a method in the ActiveJob to automatically set to queued when the job is queued....).  I think these are the right states: just "complete" isn't clear if it means success or failure, those are two different states. 

That would allow us to do things like easily search on all failed ones, etc. 

One way to do that is to use the ActiveRecord "enum" feature. https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html  By default that will store integers in the database (and want an integer column), but translate them to words in code. You can also have it store actual strings in a string column, although undocumented, with:

    enum combined_audio_derivatives_creation_status: { queued: "queued", in_progress: "in_progress" }

Then it automatically gives you methods like `sidecar.queued?` and stuff like that. 

If you think I'm being overly persnickety let me know; but I just don't like trying to figure out if something is an error or not by looking if the message starts with "Error:", the semantics should be separated from the user-displayed text like this. We might want to put an error in red, or search for all errors, etc.  It's possible there are other easier ways to do this than ActiveRecord `enum`, but it is a feature meant for it. 

It does mean if you wanted to store error info, you'd need a separate column. To begin wtih, we could just not store error info -- it should get a honeybadger report already, I think?

Next, let's get rid of the special `set_*` method, and just let the caller use ordinary `sidecar. combined_audio_derivatives_creation_status = whatever`. How do we get that to automatically set the "changed_at"? The easiest way is we can actually override the method behind that, turns out when you do `something.something = `, that's actually a method!

```ruby
def combined_audio_derivatives_creation_status=(value)
  super
  self..combined_audio_derivatives_creation_status_changed_at = DateTime.now
```

There would be cleverer ways to do that with ActiveRecord callbacks that would work even if the caller used a different method to set the status -- but your original implementation required the caller to use the "right" method anyway, so this is a good improvement. (And should have tests that when you set, it updates the `changed_at`). 

Lastly, `combined_audio_derivatives_creation_status` and `combined_audio_derivatives_creation_status_changed_at` are, phew, really long, right? I guess we were calling the basic thing `combined_audio_derivatives_creation_status` for clarity, so maybe it should match, but what do you think about just `derivatives_status` and `derivatives_status_changed_at`, there's only one derivative in the model that it could apply to!  But not sure about this, maybe the long name is good?

The next step could be making the record use activerecord callbacks to *automatically* update the "changed_at" whenever it changes, so the user

Reply:
I did almost exactly this -- of course I thought of it AFTER posting the initial PR which you reviewed. The latest version of the code is in fact adding a controller vocab. But will add queued. :)